codegen.cpp

```diff
@@ -211,13 +325,16 @@ void NodeDeclList::codegen() {
 
 void NodeFunctionCall::codegen() {
     for (size_t i = 0; i < args.size(); i++) {
-        args[i]->codegen();
-        cout << "  str x0, [sp, -16]!" << endl;
+        args[i]->codegen();  // Generate code for the argument
+        emit_push(accum_reg);  // Push the argument onto the stack
     }
-    for(int i = args.size() - 1; i >= 0; i--) {
-        cout << "  ldr x" << i << ", [sp], 16" << endl;
+    // Pop arguments into the appropriate registers (x0 to x7)
+    for (int i = args.size() - 1; i >= 0; i--) {
+        emit_pop("x" + to_string(i));
     }
-    cout << "  bl " << functionName << endl;
+    emit_function_call(functionName);  // Branch to the function
+    // Ensure the return value in x0 is moved to accum_reg (x9)
+    emit_move(accum_reg, "x0");
```

Comment:
this is wrong, and redundant. the callee already does this in its epilogue, and you would want to be moving into x0 wouldn't you? oh wait just kidding, since functions return in x0 then the caller has to copy x0 back into x9. you're right